driver-core/src/main/com/mongodb/client/model/expressions/MapExpression.java

```diff
@@ -21,45 +21,162 @@
 import static com.mongodb.client.model.expressions.Expressions.of;
 import static com.mongodb.client.model.expressions.MqlUnchecked.Unchecked.PRESENT;
 
+/**
+ * A map {@link Expression value} in the context of the MongoDB Query
+ * Language (MQL). An map is a finite, unordered a set of
+ * {@link EntryExpression entries} of a certain type, such that no entry key
+ * is repeated. It is a mapping from keys to values.
+ *
+ * @param <T> the type of the elements
+ */
 public interface MapExpression<T extends Expression> extends Expression {
 
+    /**
+     * True if {@code this} map has a value (including null) for
+     * the provided key.
+     *
+     * @param key the key.
+     * @return the resulting value.
+     */
     BooleanExpression has(StringExpression key);
```

Comment:
This method is documented differently from `DocumentExpression.has`: here it is explicitly said that keys mapped to the null values are treated as being present, but the same is neither said nor is obvious in `DocumentExpression.has`. We should further clarify the `DocumentExpression.has` method.